### PR TITLE
feat: adds updating open listing modal

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/2216)

- [x] This change addresses the issue in full

## Description
Adds a new modal when saving during editing an open listing 

## How Can This Be Tested/Reviewed?
Create a new listing that is published or edit an existing active listing 
make modifications (or don't make modifications)
click the save & exit button and the new modal should pop up

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/src/listings/Aside.tsx
Comment (dominikx96) [inline sites/partners/src/listings/Aside.tsx:94]: In other places we use `data-test-id`, so I think it would be great to follow this pattern.

File: sites/partners/src/listings/PaperListingForm/index.tsx
Comment (dominikx96) [inline sites/partners/src/listings/PaperListingForm/index.tsx:885]: It's boolean so it can be just: `open={listingIsAlreadyLiveModal}`

File: sites/partners/src/listings/PaperListingForm/sections/ListingIntro.tsx
Comment (dominikx96) [inline sites/partners/src/listings/PaperListingForm/sections/ListingIntro.tsx:72]: Same comment as in the `Aside.tsx`

**Review by dominikx96 — CHANGES_REQUESTED** [review]
@YazeedLoonat I also noticed that:

- Tests don't work in my local instance (each partners test)
- I think this modal should be only if you try to edit `active` listing, but I have it also in a listing with `draft` state

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
In testing locally: go to an open listing, hit unpublish, try to edit it & save + exit, the modal triggers in draft state

Screen Shot 2021-12-08 at 7 32 16 PM

**Comment by YazeedLoonat** [comment]
@dominikx96 & @emilyjablonski 
got this updated if you all wanna take another peep

**Comment by emilyjablonski** [comment]
LGTM from AC but I'm finding it weird that you go to hit publish or save, click OK, and then get validation errors. Is it possible to have the validation errors triggered before the modal opens?

File: backend/core/src/shared/units-transformations.ts
Comment (emilyjablonski) [inline backend/core/src/shared/units-transformations.ts:192]: 👍

File: sites/partners/cypress/integration/listing.spec.ts
Comment (emilyjablonski) [inline sites/partners/cypress/integration/listing.spec.ts:214]: What is this test testing? In isolation it seems like it makes sure the name is the name but doesn't do saving.

File: sites/partners/src/listings/Aside.tsx
Comment (emilyjablonski) [reply sites/partners/src/listings/Aside.tsx:94]: Yep we're doing that when we need a custom test id for components that have multiple of them on the page

**Comment by seanmalbert** [comment]
> LGTM from AC but I'm finding it weird that you go to hit publish or save, click OK, and then get validation errors. Is it possible to have the validation errors triggered before the modal opens?

@kathyccheng , has also brought up this issue in the past. I'm not sure there is a separate issue for this already, but I think it should be handled separately.

**Review by seanmalbert — APPROVED** [review]
LGTM, once you get @emilyjablonski's approval and tests all tests are passing.

File: sites/partners/src/listings/Aside.tsx
Comment (dominikx96) [reply sites/partners/src/listings/Aside.tsx:94]: my bad! I thought it's not a prop, looks good to me :)

**Review by dominikx96 — APPROVED** [review]

**Comment by YazeedLoonat** [comment]
@emilyjablonski can you repeep and I can merge in if you agree its okay

**Review by emilyjablonski — APPROVED** [review]
Cypress lgtm thank you! I'm going to re-run the a11y suite and when that passes you're good to merge

**Pull request merged**
